metomi/rose/rose.py

```diff
@@ -0,0 +1,323 @@
+# Copyright (C) British Crown (Met Office) & Contributors.
+#
+# This file is part of Rose, a framework for meteorological suites.
+#
+# Rose is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Rose is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Rose. If not, see <http://www.gnu.org/licenses/>.
+import argparse
+from inspect import signature
+import os
+from pathlib import Path
+import sys
+
+from pkg_resources import iter_entry_points
+
+from metomi.rose import (
+    __file__ as rose_init_file,
+    __version__,
+)
+from metomi.rose.scripts import (
+    __file__ as script_init_file,
+)
+
+
+USAGE = f'''
+Rose {__version__}
+
+Rose is a toolkit for writing, editing and running application configurations.
+'''
+
+
+# fmt: off
+DEAD_ENDS = {
+    # messages to show for commands which have been removed or renamed
+    # (ns, sub_cmd): message
+    ('rose', 'config-edit'):
+        'The Rose configuration editor has been removed, use the Cylc GUI.',
+    ('rose', 'edit'):
+        'The Rose configuration editor has been removed, use the Cylc GUI.',
+    ('rose', 'metadata-graph'):
```

Comment:
Should this still have a script if it's in this dict?

Reply:
Awaiting reimplementation, not sure we have an issue for this.

Reply:
https://github.com/metomi/rose/issues/2501